src/main/java/orionmipt/Context.java

```diff
@@ -0,0 +1,309 @@
+     * object (false, the default) or to use the result of calling
+     * toString (true).
+     */
+    public boolean useToStringAsClassName = false;
+
+    /**
+     * Toggle whether to display qualified nested class names in the
+     * label for an object from the same package as LJV (true) or
+     * to display an abbreviated name (false, the default).
+     */
+    private boolean qualifyNestedClassNames = false;
+    private boolean showPackageNamesInClasses = true;
+
+    /**
+     * Toggle whether or not to include the field name in the label for an
+     * object.  This is currently all-or-nothing.  TODO: allow this to be
+     * set on a per-class basis.
+     */
+    private boolean showFieldNamesInLabels = true;
+
+
+    private Context(Map<Object, String> classAttributeMap,
+                    Map<Object, String> fieldAttributeMap,
+                    Set<Object> pretendPrimitiveSet,
+                    Set<Object> ignoreSet) {
+        this.classAttributeMap = classAttributeMap;
+        this.fieldAttributeMap = fieldAttributeMap;
+        this.pretendPrimitiveSet = pretendPrimitiveSet;
+        this.ignoreSet = ignoreSet;
+    }
+
+    public Context() {
+        this.classAttributeMap = new HashMap<>();
+        this.fieldAttributeMap = new HashMap<>();
+        this.pretendPrimitiveSet = new HashSet<>();
+        this.ignoreSet = new HashSet<>();
+    }
+
+    public Context clone() {
+        return new Context(
+                new HashMap<>(classAttributeMap),
+                new HashMap<>(fieldAttributeMap),
+                new HashSet<>(pretendPrimitiveSet),
+                new HashSet<>(ignoreSet)
+        );
+    }
+
+    /**
+     * Set the DOT attributes for a class.  This allows you to change the
+     * appearance of certain nodes in the output, but requires that you
+     * know something about dot attributes.  Simple attributes are, e.g.,
+     * "color=red".
+     */
+    public void setClassAttribute(Class<?> cz, String attrib) {
+        classAttributeMap.put(cz, attrib);
+    }
+
+    public String getClassAtribute(Class<?> cz) {
+        return classAttributeMap.get(cz);
+    }
+
+    public Context addClassAttribute(Class<?> cz, String attrib) {
+        var result = this.clone();
+        result.setClassAttribute(cz, attrib);
+        return result;
+    }
+
+    /**
+     * Set the DOT attributes for a specific field. This allows you to
+     * change the appearance of certain edges in the output, but requires
+     * that you know something about dot attributes.  Simple attributes
+     * are, e.g., "color=blue".
+     */
+    public void setFieldAttribute(Field field, String attrib) {
+        fieldAttributeMap.put(field, attrib);
+    }
+
+    public String getFieldAttribute(Field field) {
+        return fieldAttributeMap.get(field);
+    }
+
+    public Context addFieldAttribute(Field field, String attrib) {
+        var result = this.clone();
```

Comment:
No need to copy the object here, because although its fields are final, its maps are mutable, so you can  just mutate a map here